server/src/main/java/com/thoughtworks/go/server/service/ElasticProfileService.java

```diff
@@ -59,4 +67,29 @@ public void create(Username currentUser, ElasticProfile elasticProfile, Localize
         ElasticAgentProfileCreateCommand command = new ElasticAgentProfileCreateCommand(goConfigService, elasticProfile, elasticAgentExtension, currentUser, result);
         update(currentUser, elasticProfile, result, command);
     }
+
+    public Collection<JobConfigIdentifier> getJobsUsingElasticProfile(String profileId) {
+        if (findProfile(profileId) == null) {
+            throw new RecordNotFoundException(String.format("Elastic profile with id '%s' does not exist.", profileId));
+        }
+
+        final List<PipelineConfig> allPipelineConfigs = goConfigService.getAllPipelineConfigs();
+        final Set<JobConfigIdentifier> jobsUsingElasticProfile = new HashSet<>();
+
+        for (PipelineConfig pipelineConfig : allPipelineConfigs) {
```

Comment:
@bdpiparva @vishaldevgire - wouldn't this be a very slow operation? we are iterating over all-pipelines/stages/jobs and checking if a job required an elastic profile.

Can we cache this(a map of profile-to-jobs) under `GoCache` and rely on `PipelineConfigChangeListener` and `ElasticProfileChangeListener`(Need to create this) to modify cached information.

Reply:
@GaneshSPatil - Config is already cached. We tried with 4k pipelines and 200 profiles. We did not notice any lag in response.